app/src/main/java/com/github/polypoly/app/SignInActivity.kt

```diff
@@ -0,0 +1,241 @@
+import androidx.compose.ui.res.painterResource
+import androidx.compose.ui.unit.dp
+import com.firebase.ui.auth.AuthUI
+import com.firebase.ui.auth.FirebaseAuthUIActivityResultContract
+import com.github.polypoly.app.base.game.rules_and_lobby.GameLobby
+import com.github.polypoly.app.base.game.rules_and_lobby.GameMode
+import com.github.polypoly.app.base.game.rules_and_lobby.GameRules
+import com.github.polypoly.app.base.user.Skin
+import com.github.polypoly.app.base.user.Stats
+import com.github.polypoly.app.base.user.User
+import com.github.polypoly.app.global.GlobalInstances.Companion.currentUser
+import com.github.polypoly.app.global.GlobalInstances.Companion.isSignedIn
+import com.github.polypoly.app.global.GlobalInstances.Companion.remoteDB
+import com.github.polypoly.app.global.Settings.Companion.DB_GAME_LOBIES_PATH
+import com.github.polypoly.app.global.Settings.Companion.DB_USERS_PROFILES_PATH
+import com.github.polypoly.app.menu.kotlin.GameMusic
+import com.github.polypoly.app.network.RemoteDB
+import com.github.polypoly.app.ui.theme.PolypolyTheme
+import com.google.firebase.auth.FirebaseAuth
+import com.google.firebase.database.ktx.database
+import com.google.firebase.ktx.Firebase
+import java.util.concurrent.CompletableFuture
+
+/**
+ * This activity is holds the sign up flow. It will be displayed whenever no user is signed in.
+ */
+class SignInActivity : ComponentActivity() {
+
+    private var firebaseAuth: FirebaseAuth? = null
+    private var mAuthListener: FirebaseAuth.AuthStateListener? = FirebaseAuth.AuthStateListener {}
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+
+        // Global initialization of the database
+        val db = Firebase.database
+        remoteDB = RemoteDB(db, "live")
+        // addFakeDataToDB() // < -- uncomment this line to add fake data to the DB
+
+        firebaseAuth = FirebaseAuth.getInstance()
+        isSignedIn = false
+        if (firebaseAuth!!.currentUser != null) {
+            launchWelcome()
+        }
+        setContent { SignInContent() }
+    }
+
+    override fun onResume() {
+        super.onResume()
+        firebaseAuth!!.addAuthStateListener(mAuthListener!!)
+    }
+
+    override fun onStop() {
+        super.onStop()
+        firebaseAuth!!.removeAuthStateListener(mAuthListener!!)
+    }
+
+    /**
+     * starts the WelcomeActivity and sets the isSignedIn flag to true
+     */
+    private fun launchWelcome(){
+        val welcomeActivityIntent = Intent(this, WelcomeActivity::class.java)
+        isSignedIn = true
+        startActivity(welcomeActivityIntent)
+        finish()
+    }
+
+    /**
+     * launched when the sign-in flow is wanted to be started
+     */
+    private val signInLauncher = registerForActivityResult(
+        FirebaseAuthUIActivityResultContract()
+    ) {}
+
+    @Composable
+    fun SignInContent() {
+        GameMusic.setSong(LocalContext.current, R.raw.mocksong)
```

Comment:
Why is the music launched in the sign in activity? This seems a weird Software Engineering practice

Reply:
You are right it was not intended. I think I have put it there because Signing activity is the main activity now, but it makes sense for the music to launch only when we reach the menu.